build.js

```diff
@@ -51,6 +52,56 @@ StyleDictionary.registerTransform({
   },
 });
 
+StyleDictionary.registerFormat({
+  name: `typescript/color-declaration`,
+  formatter: ({ dictionary, file }) => {
+    return (
+      fileHeader({ file }) +
+      Object.entries(dictionary.properties.colors)
+        .map((tokens) => {
+          const colorObj = tokens[0];
+          const filtered = dictionary.allTokens.filter(
+            (token) => token.attributes.type === colorObj
+          );
+          return (
+            `export const ${colorObj} : { ` +
+            filtered.map((token) => {
+              return `${token.name} : ${getTypeScriptType(token.value)}`;
+            }) +
+            `};`
+          );
+        })
+        .join(`\n`)
+    );
+  },
+});
+
+StyleDictionary.registerFormat({
+  name: `javascript/colors`,
+  formatter: ({ dictionary, file }) => {
+    return (
+      fileHeader({ file }) +
+      `module.exports = {` +
+      Object.entries(dictionary.properties.colors)
+        .map((tokens) => {
+          const colorObj = tokens[0];
+          const filtered = dictionary.allTokens.filter(
+            (token) => token.attributes.type === colorObj
+          );
+          return (
+            `${colorObj} : { ` +
+            filtered.map((token) => {
+              return `${token.name} : "${token.value}"`;
+            }) +
+            `},`
+          );
+        })
```

Comment:
This map function and the one on line 61 are baaasically the same thing, I wonder if we can create a reusable snippet for this, especially if we need to change it. Only difference is if it's getting a TS value instead of JS

Reply:
100% - def want to clean it up and streamline it where possible